ostd/src/mm/io.rs

```diff
@@ -708,9 +706,7 @@ impl<'a> VmWriter<'a, Infallible> {
     ///
     /// # Safety
     ///
-    /// `ptr` must be [valid] for writes of `len` bytes during the entire lifetime `a`.
-    ///
-    /// [valid]: crate::mm::io#safety
+    /// `ptr` must be valid for writes of `len` bytes during the entire lifetime `a`.
```

Comment:
The term "valid" is very obscure and can mean many different things depending on the context. We need to give it a formal definition.

I think we can make the `io` module public and remove the unnecessary public re-exports of readers and writers. At least, I fail to see the value of those public re-exports.
https://github.com/asterinas/asterinas/blob/5e3e23bf7c5130a9251b265c5377d2e21769097c/ostd/src/mm/mod.rs#L35-L38

Reply:
The `io` module is made `pub` now. But the path cleanup is not included.